akka-actor-typed-tests/src/test/scala/akka/actor/typed/DispatcherSpec.scala

```diff
@@ -0,0 +1,83 @@
+package akka.actor.typed
+
+import akka.actor.ActorCell
+import akka.actor.testkit.typed.scaladsl.{ LogCapturing, ScalaTestWithActorTestKit }
+import akka.actor.typed.internal.adapter.ActorContextAdapter
+import akka.actor.typed.scaladsl.AskPattern._
+import akka.actor.typed.scaladsl.Behaviors
+import akka.dispatch._
+import org.scalatest.wordspec.AnyWordSpecLike
+
+class DispatcherSelectorSpec extends ScalaTestWithActorTestKit("""
+    specific-dispatcher {
+      type = Dispatcher
+      executor = "thread-pool-executor"
+    }
+    specific-mailbox {
+      mailbox-type = "akka.dispatch.NonBlockingBoundedMailbox"
+      mailbox-capacity = 4
+    }
+    specific-dispatcher-with-mailbox {
+      type = Dispatcher
+      executor = "thread-pool-executor"
+      mailbox-type = "akka.dispatch.NonBlockingBoundedMailbox"
+      mailbox-capacity = 4
+    }
+  """) with AnyWordSpecLike with LogCapturing {
+
+  case class WhatsYourDispatcherAndMailbox(replyTo: ActorRef[(MessageDispatcher, MessageQueue)])
+
+  private def behavior: Behavior[WhatsYourDispatcherAndMailbox] =
+    Behaviors.setup { context =>
+      Behaviors.receiveMessage[WhatsYourDispatcherAndMailbox] {
+        case WhatsYourDispatcherAndMailbox(replyTo) =>
+          val (dispatcher, mailbox) = context match {
+            case adapter: ActorContextAdapter[_] =>
+              adapter.classicActorContext match {
+                case cell: ActorCell =>
+                  (cell.dispatcher, cell.mailbox.messageQueue)
+                case unexpected => throw new RuntimeException(s"Unexpected: $unexpected")
+              }
+            case unexpected => throw new RuntimeException(s"Unexpected: $unexpected")
+          }
+
+          replyTo ! (dispatcher, mailbox)
+          Behaviors.stopped
+      }
+    }
+
+  "The Dispatcher selectors" must {
+
+    "default to unbounded" in {
+      val actor = spawn(behavior)
+      val (dispatcher, mailbox) = actor.ask(WhatsYourDispatcherAndMailbox.apply).futureValue
+      dispatcher.id eq Dispatchers.DefaultDispatcherId
```

Comment:
```
dispatcher shouldBe Dispatchers.DefaultDispatcherId
```
?

Reply:
and in more places below

Reply:
> ```
> dispatcher shouldBe Dispatchers.DefaultDispatcherId
> ```
> 
> ?

The types was mismatch, but I've already fixed it when constructing the return result.